epochStart/bootstrap/process.go

```diff
@@ -462,6 +459,24 @@ func (e *epochStartBootstrap) requestAndProcessing() (Parameters, error) {
 	return parameters, nil
 }
 
+func (e *epochStartBootstrap) saveSelfShardId() {
+	if e.baseData.shardId == core.AllShardId {
+		destShardID := core.MetachainShardId
+		if e.destinationShardAsObserver != "metachain" {
+			var destShardIDUint64 uint64
+			destShardIDUint64, err := strconv.ParseUint(e.destinationShardAsObserver, 10, 64)
+			if err == nil && destShardIDUint64 < uint64(e.baseData.numberOfShards) {
```

Comment:
Both if and else do the same thing. Please refactor. Probably forgot to remove `destShardID = e.genesisShardCoordinator.SelfId()` from if.

Reply:
done